src/sentry/static/sentry/app/views/onboarding/platform.jsx

```diff
@@ -0,0 +1,158 @@
+import {css} from 'react-emotion';
+import PropTypes from 'prop-types';
+import React from 'react';
+
+import {addErrorMessage} from 'app/actionCreators/indicator';
+import {createProject} from 'app/actionCreators/projects';
+import {stepPropTypes} from 'app/views/onboarding/wizardNew';
+import {t, tct} from 'app/locale';
+import Button from 'app/components/button';
+import PlatformPicker from 'app/components/platformPicker';
+import ProjectActions from 'app/actions/projectActions';
+import SentryTypes from 'app/sentryTypes';
+import space from 'app/styles/space';
+import withApi from 'app/utils/withApi';
+import withTeams from 'app/utils/withTeams';
+
+const platformListStyle = css`
+  max-height: 420px;
+  overflow-y: scroll;
+  /* Needed to fix overflow cropping of the de-select button */
+  margin-top: -${space(2)};
+  padding-top: ${space(2)};
+`;
+
+class OnboardingPlatform extends React.Component {
+  static propTypes = {
+    ...stepPropTypes,
+    api: PropTypes.object,
+    teams: PropTypes.arrayOf(SentryTypes.Team),
+  };
+
+  state = {
+    /**
+     * This will be flipped to true immedaitely before  creating the first
+     * project. We use state here to avoid the intermittent prop value where
+     * the project is created but the store hasn't propagated its value to the
+     * component yet, leaving a brief period where the button will flash
+     * between labels / disabled states.
+     */
+    firstProjectCreated: false,
+    /**
+     * `progressing` indicates that we are moving to the next step. Again, this
+     * is kept as state to avoid intermittent states causing flickering of the
+     * button.
+     */
+    progressing: false,
+  };
+
+  componentDidUpdate(prevProps, prevState) {
+    if (prevProps.active && !this.props.active) {
+      // eslint-disable-next-line react/no-did-update-set-state
+      this.setState({progressing: false});
+    }
+  }
+
+  get hasFirstProject() {
+    return this.props.project || this.state.firstProjectCreated;
+  }
+
+  get contineButtonLabel() {
+    if (this.state.progressing) {
+      return t('Creating Project...');
+    }
+    if (!this.hasFirstProject) {
+      return t('Create Project');
+    }
+    if (!this.props.active) {
+      return t('Project Created');
+    }
+    return t('Setup Your Project');
+  }
+
+  async createFirstProject(platform) {
+    const {api, orgId, teams} = this.props;
+
+    if (this.hasFirstProject) {
+      return;
+    }
+
+    if (teams.length < 1) {
+      return;
+    }
+
+    this.setState({firstProjectCreated: true});
+
+    try {
+      const data = await createProject(api, orgId, teams[0].slug, orgId, platform);
```

Comment:
`orgId` is passed twice?

Reply:
It’s used as the name of the project, since we don’t ask for their first project name. Probably would be better for createProject to take an object so we can use arg keys.